### PR TITLE
Implement FOR..NEXT

### PR DESCRIPTION
**Pull request by boxdot**
Implementation unfolds FOR..NEXT blocks into a flat statements by replacing them with equivalent LET/IFTHEN/GOSUB statements.

### CODE REVIEW

**Comment by boxdot** [comment]
Resolves #16

File: src/ast.rs [hunk 1]
```diff
@@ -68,39 +65,202 @@ impl<'a> Program<'a> {
...
-                            line_number: *line_number,
+                    self.index_last_block()?;
+                }
+                // Flatten FOR...NEXT by replacing FOR statement with LET preamble and
+                // IF statement, NEXT statement with GOSUB.
+                Block::For {
+                    for_line,
+                    blocks,
+                    next_line,
+                } => {
```
Comment (jeschkies) [inline src/ast.rs]: What do you think about factoring out the statements? I think this would make reading it easier.

File: src/ast.rs [hunk 2]
```diff
@@ -68,39 +65,202 @@ impl<'a> Program<'a> {
...
                         });
                     }
+
+                    // LET .. = limit
+                    let limit = NumericVariable::Limit {
+                        line_number: for_line.line_number,
+                    };
+                    self.blocks.push(Block::Line {
+                        line_number: internal_line_numer,
+                        statement_source: Span::new(CompleteStr("")),
```
Comment (jeschkies) [inline src/ast.rs]: This is kind of misleading. What do you think about using an optional for the `statement_source` or a `Block::Phony`?

File: src/ast.rs (hunk 1 above)
Comment (boxdot) [reply src/ast.rs]: I will create an issue for that. You are right, this needs to be cleaned up.

File: src/ast.rs (hunk 2 above)
Comment (boxdot) [reply src/ast.rs]: Ah, this is good proposal. Will do it in sub sequential review.

**Pull request merged**